server/api/utils/singletons/k8s.py

```diff
@@ -125,6 +125,99 @@ def list_pods(self, namespace=None, selector="", states=None):
                 items.append(i)
         return items
 
+    @raise_for_status_code
+    def list_pods_paginated(self, namespace=None, selector="", states=None):
```

Comment:
Feels to me like a lot of these 2 functions can be generalized into a `list_paginated` method that receives a list method callback, and can be later reused by `list_pods_paginated`, `list_crds_paginated` and any other resource we want.

Reply:
I'm not sure I agree. There are differences with how to parse the result which needs to be part of the callback and it turns out I'm extracting 90% of the method to the callback. If you have some other Idea lets discuss offline but in general I think that the common part is not big enough to make this refactor.